libsawtooth/src/journal/genesis.rs

```diff
@@ -0,0 +1,638 @@
+/*
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use std::convert::TryFrom;
+use std::error::Error;
+use std::fs::{remove_file, File};
+use std::io::Read;
+use std::path::PathBuf;
+use std::sync::mpsc::{channel, Receiver, Sender};
+
+use cylinder::Signer;
+use transact::execution::executor::ExecutionTaskSubmitter;
+use transact::protocol::receipt::TransactionResult;
+use transact::scheduler::{BatchExecutionResult, SchedulerError, SchedulerFactory};
+use transact::state::{StateChange, Write};
+
+use crate::journal::block_manager::BlockManager;
+use crate::journal::chain::ChainObserver;
+use crate::protocol::block::BlockBuilder;
+use crate::protocol::genesis::GenesisData;
+use crate::protos::transaction_receipt::TransactionReceipt;
+use crate::protos::FromBytes;
+use crate::state::merkle::CborMerkleState;
+use crate::state::settings_view::SettingsView;
+use crate::state::state_view_factory::StateViewFactory;
+
+use super::{chain::ChainReader, chain_id_manager::ChainIdManager, NULL_BLOCK_IDENTIFIER};
+
+const GENESIS_FILE: &str = "genesis.batch";
+
+/// Used to combine errors and `Batch` results returned from the `Scheduler`
+enum SchedulerEvent {
+    Result(BatchExecutionResult),
+    Error(SchedulerError),
+    Complete,
+}
+
+/// Builder for creating a `GenesisController`
+#[derive(Default)]
+pub struct GenesisControllerBuilder {
```

Comment:
Can you move the builder to its own file? It's a pretty big chunk of code that's irrelevant to how the genesis controller actually works.